src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1054,30 +1056,44 @@ event::ptr primitive_inst::update_weights() {
     return nullptr;
 }
 
-static bool user_requesting_mem_reuse_false(const program_node& node) {
-    for (auto& user : node.get_users()) {
-        if (user->is_dynamic())
-            return true;
-        if ((user->get_selected_impl() != nullptr) && (user->get_selected_impl()->can_reuse_memory == false)) {
-            return true;
-        } else if (user->get_selected_impl() == nullptr) {
-            if (user_requesting_mem_reuse_false(*user)) {
+static bool user_requesting_mem_reuse_false(const program_node& node, const primitive_inst* prim, bool runtime_alloc) {
+    if (prim == nullptr || runtime_alloc == false) {
+        for (auto& user : node.get_users()) {
+            if ((user->get_selected_impl() != nullptr) && (user->get_selected_impl()->can_reuse_memory == false)) {
                 return true;
+            } else if (user->get_selected_impl() == nullptr) {
+                if (user_requesting_mem_reuse_false(*user, nullptr, runtime_alloc)) {
+                    return true;
+                }
+            }
+        }
+    } else {
+        for (auto& user : prim->get_user_insts()) {
+            if ((user->get_impl() != nullptr) && (user->get_impl()->can_reuse_memory == false)) {
+                return true;
+            } else if (user->get_impl() == nullptr) {
+                if (user_requesting_mem_reuse_false(user->get_node(), user.get(), runtime_alloc)) {
+                    return true;
+                }
             }
         }
```

Comment:
ah... I just realized that the "user impl" is not decided yet when allocating current inst's output. 
I think maybe we need to adjust the conditions further. Let's think about it later...

Reply:
And now, I think we don't need "user_requesting_mem_reuse_false" at all because it should be dealt by need_reset_output_memory, if the user requires memory reset. (e.g., we can add this function to some types of reduce)

Reply:
I updated it not to use `user_requesting_mem_reuse_false` in the case of dynamic models, as you guided. Thank you.